app/models/concerns/host_params.rb

```diff
@@ -21,7 +21,13 @@ def inherited_params_hash
     end
 
     def non_inherited_params_hash
-      params_to_hash(host_parameters.authorized(:view_params))
+      items = []
+      items << host_parameters.authorized(:view_params)
+      # Add transient host parameters to the array
+      # :authorized removes records that are not saved in the db
+      items << host_parameters.reject { |hp| hp.persisted? } if User.current.can?(:view_params)
+
+      params_to_hash(items.flatten.uniq { |hp| hp.name })
```

Comment:
Without looking at the overall picture some things that jump out to me: avoid a list that's flattened later (which i assume has a better memory usage) and use RuboCop suggestions to have shorter calls.
```suggestion
      items = host_parameters.authorized(:view_params).to_a
      # Add transient host parameters to the array
      # :authorized removes records that are not saved in the db
      items += host_parameters.reject(&:persisted?) if User.current.can?(:view_params)

      params_to_hash(items.uniq(&:name))
```